src/components/ProductCard.tsx

```diff
@@ -1,89 +1,86 @@
-      <div className="bg-gray-100 group flex flex-col relative justify-center border-4 border-white">
-        <div className="z-30">
-          <Transition
-            show={!isCardFlipped}
-            className=""
-            enter="transform transition ease-in-out duration-300"
-            enterFrom="opacity-0 -translate-x-0"
-            enterTo="opacity-100 translate-x-0"
-            leave="transform transition ease-in-out duration-300"
-            leaveFrom="opacity-0 translate-x-0"
-            leaveTo="opacity-100 translate-x-0"
-          >
-            <button
-              className="group-hover:visible flex flex-row absolute top-0 right-0 py-1 px-2 text-xs bg-gray-300 z-20 "
-              onClick={toggleIsFlippedCard}
-            >
-              SHOW MORE
-              <Image
-                className="flex relative left-1"
-                src="/icons8-plus-50.png"
-                alt="X"
-                width={14}
-                height={14}
-                style={{ height: "15px", width: "15px" }}
-              />
-            </button>
-          </Transition>
-          <Transition
-            show={isCardFlipped}
-            className=""
-            enter="transform transition ease-in-out duration-300"
-            enterFrom="opacity-0 -translate-x-0"
-            enterTo="opacity-100 translate-x-0"
-            leave="transform transition ease-in-out duration-300"
-            leaveFrom="opacity-0 translate-x-0"
-            leaveTo="opacity-100 translate-x-0"
-          >
-            <button
-              className="group-hover:visible flex flex-row absolute top-0 right-0 py-1 px-2 text-xs bg-gray-300 z-20"
-              onClick={toggleIsFlippedCard}
-            >
-              CLOSE
-              <Image
-                className="rotate-45 flex relative left-1"
-                src="/icons8-plus-50.png"
-                alt="X"
-                width={14}
-                height={14}
-                quality={100}
-              />
-            </button>
-          </Transition>
-        </div>
+      <div
+        key={key}
+        className="bg-gray-100 group flex flex-col relative justify-center border-4 border-white"
+      >
+        <ProductCardShowMoreButton
+          isCardFlipped={isCardFlipped}
+          changeIsCardFliped={changeIsCardFliped}
+          changeImage={changeImage}
+          content={content}
+        />
+       </div>
 
-        <div className="flex flex-col w-[413.33px] h-[508px] items-center relative">
+        <div className="flex flex-col w-[413.33px] h-[508px] items-center relative bg-gray-100">
           <div className="top-10 relative">
-            <Image src={cardImage} height={300} width={300} alt="cardImage" />
+            {isCardFlipped ? (
+              <div className="[transform:rotateY(180deg)] ease-in-out [transform-style-:preserve-3d] [backface-visblity:hidden] transition-all duration-1000">
```

Comment:
I'm unfamiliar with the use of [] here, what is going on here and why do we need these for animations?

Reply:
these allow me to write an arbitrary css classes in tailwind, as the classes wrapped in square brackets are classes that tailwind does not have as default.